### DISCUSSION TITLE
Having trouble adding image assets

### DISCUSSION THREAD

**Discussion by evanb2** [crxjs/chrome-extension-tools#225] [Q&A] [answered]
Didn't want to create an issue cause I'm not 100% I have my project setup right, but none of my images are getting bundled into the build.

`package.json`
```json
"vite": "^2.7.2",
"vue": "^3.2.25",
"rollup-plugin-chrome-extension": "^4.0.1-16",
```

`manifest.json`
```json
  "action": {
    "default_popup": "src/popup/index.html",
    "default_icon": {
      "16": "src/assets/images/logo_16.png",
      "32": "src/assets/images/logo_32.png",
      "48": "src/assets/images/logo_48.png",
      "128": "src/assets/images/logo_128.png"
    }
  },
  "icons": {
    "16": "src/assets/images/logo_16.png",
    "32": "src/assets/images/logo_32.png",
    "48": "src/assets/images/logo_48.png",
    "128": "src/assets/images/logo_128.png"
  },
  "web_accessible_resources": [
    {
      "resources": [
        "src/assets/images/Logo-Chrome@2x.svg",
        "src/assets/images/Logo-Chrome-Tab@2x.svg"
      ],
      "matches": ["https://*/*"]
    }
  ]
```
`vite.config.js`
```js
import vue from '@vitejs/plugin-vue'
import { resolve } from 'path'
import { defineConfig } from 'vite'
import { chromeExtension } from 'rollup-plugin-chrome-extension'
import manifest from './manifest.json'

// https://vitejs.dev/config/
export default defineConfig({
  resolve: {
    alias: {
      '@': resolve(__dirname, 'src'),
    },
  },
  plugins: [chromeExtension({ manifest }), vue()],
})

```

Folder structure:
```
- src/
  |- assets/
    |- images/
       - logo_16.png
       - logo_32.png
       - logo_48.png
       - logo_128.png
- manifest.json
- vite.config.js
```

Docs say this is suppose to happen automagically from the manifest file. I'm not seeing any issues regarding this so assuming I'm just missing something lol.

Thanks for any help!

**Comment by jacksteamdev** [comment, marked as answer]
@evanb2 Thanks for reporting this! Looks like this was an oversight in the most recent rewrite. I think it's a nice feature, so I'm converting this to an issue.

### Workaround

Try putting images in the public folder for now, that should work in production. Once #224 lands it will work in development. 

```jsonc
// manifest.json
{
  "icons": {
    "16": "assets/images/logo_16.png",
    "32": "assets/images/logo_32.png",
    "48": "assets/images/logo_48.png",
    "128": "assets/images/logo_128.png"
  },
}
```
```
- src/
- public/
  |- assets/
    |- images/
       - logo_16.png
       - logo_32.png
       - logo_48.png
       - logo_128.png
- manifest.json
- vite.config.js
```

**Reply by evanb2** [reply]
Nice! AFAICT this also works in `development` (running `vite dev`)

**Reply by evanb2** [reply]
@jacksteamdev Thanks for a speedy response! This plugin is an absolute lifesaver, been trying to migrate from a custom Webpack config to Vite and this was the missing piece.

**Comment by blujedis** [comment]
So this still seems to be relevant. When I specify subfolders in the public directory `public/images/logo_16.png` the output will be `dist/images/logo_16.png`.  The problem with this is any images at the root of `public` are output as `public/logo_16.png`. 

Probably not what you want. Perhaps there's some config I'm missing but that was the behavior I was seeing. 

The resolution seems to be to add the `assets` folder under the `public` directory then all assets are combined in the `dist/assets` directory.

Then in your manifest you would use:

```ts
  icons: {
    16: 'assets/icons/light/icon16.png',
    32: 'assets/icons/light/icon32.png',
    48: 'assets/icons/light/icon48.png',
    128: 'assets/icons/light/icon128.png'
  }
```

My .02 would be to change `public` to `static` and then output as is. That or do away with the public folder and just use `src/assets`. You could have a subfolder there called static also so that it is clear which files are just pass through if you will. 

Feel free to tell me to pound sand if there's something I've missed :)
